config-controller/main.go

```diff
@@ -45,7 +47,9 @@ import (
 
 var (
 	scheme   = runtime.NewScheme()
-	setupLog = ctrl.Log.WithName("setup")
+	setupLog = logging.LoggerForModule()
+	// the default resync period for the manager cache is 10 hours if unspecified. We set it
+	resyncPeriod = 4 * time.Hour
```

Comment:
Also, did you consider the controller-runtime suggestion to requeue each policy CR using the `Result{requeueAfter: x}` return object instead of relying on resync of the cache?

Reply:
We could do that for successful updates and creates, but pretty sure if the config-controller restarts that queue is lost. So the resync period setting is necessary. The comment you linked to does not seem to explain how those interact together - what if both are set.

Reply:
Just for reference, here is the code comment:  https://github.com/kubernetes-sigs/controller-runtime/blob/v0.18.5/pkg/cache/cache.go#L145-L170

> if the config-controller restarts that queue is lost

True, but a fully resync happens at startup anyways

> what if both are set

I think it is the union of all reconciliation events.  So there would just be more reconciles for any given policy, both when the requeue timeout occurs and also during the cache resync.  

(Note that the "resync" in effect just adds one reconciliation event to the queue for every policy that the informer is watching at that given moment.)

Reply:
I set the requeue after to 4 hours as well, not sure if that is what you had in mind as the duration.

Reply:
I don't think I'm necessarily convinced we should do the requeue instead of modifying the resync. The code comment seemed to suggest it was better, but it doesn't really explain why.  I just thought it was interesting that it suggested that and might be worth considering.

If I had to guess, the upside of explicit requeues would be to help spread the load of reconciles so that they don't all happen at once (like they do during a resync).

I think it would make sense to choose one or the other: either requeue all policies, or configure resync.  That said, as I mentioned above it doesn't really hurt much to do both (just some extra reconciles).  It might be a bigger deal if there were thousands of objects reconciling, but it shouldn't be an issue at our expected scale.

> not sure if that is what you had in mind as the duration.

I don't have a strong opinion here.  Four hours seems fine to me, though imo we could make the duration smaller without a lot of additional overhead.

Reply:
In that light, here's what I propose: Skip the requeue after completely for now. Set the resync period to 4 hours. Now we can expect, in terms of debug, and logs, to see a whole set of reconciles on 1) startup 2) at the end of resync period 3) and a bunch of reconcile events based on source code repo activity.